src/App.tsx

```diff
@@ -122,14 +133,27 @@ export default function App() {
           </IconButton>
         </div>
         <Divider />
-
-        <Upload />
-        <BasicSettings />
-        <Divider />
-        <RegionSettings
-          regions={[]}
-          selectableRegions={["maxilla", "jawbone"]}
-        />
+        {filesUploaded ? (
+          <>
+            <BasicSettings />
+            <Divider />
+            <RegionSettings
+              regions={[]}
+              selectableRegions={["maxilla", "jawbone"]}
+            />
+            <Divider />
+            <Button
+              color="secondary"
+              variant="contained"
+              className={classes.resetButton}
+              onClick={() => dispatch(resetUploadedFiles())}
+            >
+              Reset Uploaded Files
+            </Button>
```

Comment:
Html tego komponentu trochę się rozrasta. 
To ma dwie wady:
1. trudniej się czyta
2. całość się rerenderuje w przypadku jeśli coś w komponencie się zmieni. (o ile dobrze pamiętam, ale spróbuję jeszcze znaleźć jakieś żródło.)


Dlatego ja bym rozbił ten komponent podkomponenty:
1. Drawer
2. Toolbar
3. Canvas


Ponadto biorąc pod uwagę, że to jest `App.tsx`, czyli miejsce w którym może się pojawić dość sporo kodu związanego z konfigurację różnego Middleware. 

Wiem, że to nie jest do końca twój kod, ale teraz rzuciło mi się w oczy.